### DISCUSSION TITLE
Some Ideas? Not sure where to put this.

### DISCUSSION THREAD

**Discussion by jdroo** [OpenRCT2/OpenRCT2#15752] [Feature requests]
_Not sure where to put Ideas. so putting it here!_

**----These are some Ideas for the cheat menu.----**

-Check box for "Always mowed grass" and "Always clear grass".
-Check boxes for locking Hunger, Thirst,  Nausea, and bathroom to 0.
-Check boxes for locking Energy, and Happiness to 100%.
-Check box to lock guests money at 1000$.
-Check box to make guests always walk at their max speed. [speedy guests]
-Have a number box to set the size of the tram spawned.

-Random color check box for staff when spawning in mechanics, guards, and Handymen.
-Random color check box for balloon stalls, and other merchandise that can have a color set, like hats, ect. So can have random colors naturally, without having to have multiple stands.

-Tab for all the special named guests [Easter eggs] that rct and rct2 have, to spawn them in without having to name anyone. [or maybe just a check box to spawn in "special guests" randomly on their own].

**----Other random ideas.----**

-Fix it so if "automatically place staff" is on, that it does not display the pop up window each time it sets them down.
-Fix it so when you have "invisible people" on, it does not show them in the other pop up windows still.

-An option to zoom in more than usual with the mouse wheel. [I have shortcut keys to use the scale factor this way, but it zooms the UI in too, kinda annoying, a cleaner zoom would be nice].

-Option to choose a custom mp3, or whatever format rct2 uses, to set custom music on rides on the fly, without having to move any files on your computer. Just set the songs file path, and you're good to go! [not sure if possible].

-A separate volume slider for just the guests sounds volume, so can still hear rides.

-

**Comment by Lastorder-DC** [comment]
Since plugin system is already implemented, new functions will likely not implemented to game itself.
If API is needed for plugin implement, you can always ask at discord, or open new issue for that.

**Comment by duncanspumpkin** [comment]
These all look like plugin ideas. We are wanting to remove things from the cheats window and move them to plugins.

**Comment by ocalhoun6** [comment]
>-Fix it so if "automatically place staff" is on, that it does not display the pop up window each time it sets them down.

I wouldn't want this one. 

_Sometimes_ when I'm hiring staff, I really want them placed randomly, but a lot of times, the first thing I do after the staff (especially handyman) is hired is that I tweezer him to the spot where I want him and then set a patrol area ... which is good to have his window open for.

>-Fix it so when you have "invisible people" on, it does not show them in the other pop up windows still.

Sometimes it might be nice to have global view modes applied to smaller windows as well ... but then other times you wouldn't want that. (Just because you're building something underground in the main view doesn't necessarily mean you want all pop-up windows to show underground view as well.)

**Comment by jdroo** [comment]
How bout this then. For multiplayer, I've noticed when you have a server that allows cheats, it allows all the cheats, it would be nice to specify what cheats you want allowed, and disable ones you don't want people using, Like when someone joins your server and spams a macro to win the scenario over and over, that causes all open windows to close, making it almost impossible to kick the spammer.

**Comment by Margen67** [comment]
Related: #1438

**Comment by Gymnasiast** [comment]
- Check box for "Always mowed grass" and "Always clear grass".
- Check boxes for locking Hunger, Thirst, Nausea, and bathroom to 0.
- Check boxes for locking Energy, and Happiness to 100%.
- Check box to lock guests money at 1000$.
- Check box to make guests always walk at their max speed. [speedy guests]
- Have a number box to set the size of the tram spawned.

These are all stuff for a plug-in, I’d say.

- Random color check box for staff when spawning in mechanics, guards, and Handymen.

Not possible unless I’m very mistaken, they share the same colour.

- Random color check box for balloon stalls, and other merchandise that can have a color set, like hats, ect. So can have random colors naturally, without having to have multiple stands.

#14381 

- Tab for all the special named guests [Easter eggs] that rct and rct2 have, to spawn them in without having to name anyone. [or maybe just a check box to spawn in "special guests" randomly on their own].

_Definitely_ a plug-in thing.

----Other random ideas.----

- Fix it so if "automatically place staff" is on, that it does not display the pop up window each time it sets them down.
- Fix it so when you have "invisible people" on, it does not show them in the other pop up windows still.

Could be considered.

- An option to zoom in more than usual with the mouse wheel. [I have shortcut keys to use the scale factor this way, but it zooms the UI in too, kinda annoying, a cleaner zoom would be nice].

Now implemented.

- Option to choose a custom mp3, or whatever format rct2 uses, to set custom music on rides on the fly, without having to move any files on your computer. Just set the songs file path, and you're good to go! [not sure if possible].

The new save format supports music objects.

- A separate volume slider for just the guests sounds volume, so can still hear rides.

Sounds overkill to me.

**Comment by Broxzier** [comment]
>     Random color check box for staff when spawning in mechanics, guards, and Handymen.
>
>Not possible unless I’m very mistaken, they share the same colour.

It is possible, it's just that when changing the colour it applies it to all staff members:
https://github.com/OpenRCT2/OpenRCT2/blob/98f9f2490908e864e548ddc0fdee0ee16698e671/src/openrct2/actions/StaffSetColourAction.cpp#L58-L65

This does seem more fitting as a plug-in. AFAIK, the colours can already be changed by plug-ins.

**Comment by ZehMatt** [comment]
> How bout this then. For multiplayer, I've noticed when you have a server that allows cheats, it allows all the cheats, it would be nice to specify what cheats you want allowed, and disable ones you don't want people using, Like when someone joins your server and spams a macro to win the scenario over and over, that causes all open windows to close, making it almost impossible to kick the spammer.

You could create a new user group that disallows cheats.

**Comment by tcek** [comment]
If everything is being moved from the cheats menu to plugins, how do I go about getting plugins up? Don't know how to do it.
